### ISSUE TITLE
feature engineering

### ISSUE THREAD

**Issue by autokad** [cognoma/machine-learning#1]
• Feature Engineering
o   Feature Transformations
 Log
 Square
 Inverse
 Percentile
 ZScore
o   Feature Creation
 Interactions (+ \* - )
 LDA (Latent Dirichlet Allocation)
• Feature Reduction (Selection / Extraction)
o   Stepwise Regression
o   RFE (Recursive Feature Elimination)
o   PCA (Principle Component Analysis)
o   LDA (Latent Dirichlet Allocation) 
o   Linear Discriminate Analysis -Also LDA =/
o   Genetic Algorithms
o   Wrapper Methods
• Algorithms
o   Linear Regression
 Ridge Regression
 LASSO
 Elastic Net
 OLS (Ordinary Least Squares)
o   Logistic Regression
o   SVM (Support Vector Machine)
 RBF Kernel (Radial Basis Function)
 Polynomial & Linear Kernel
 Histogram Kernel
o   Random Forest
o   Adaboost
o   Logitboost
o   KNN (K Nearest Neighbors)
o   Naïve Bayes
o   K Means
o   Perceptron
o   Neural Nets
o   GBM (Gradient Boosting Machines)

**Comment by autokad** [comment]
These can be broken down as functions we need to implement in django and also something we an all try to get a handle of machine learning, feel free to add to the list

**Comment by yl565** [comment]
- Feature Reduction (Selection / Extraction)
  - Canonical correlation analysis

**Comment by dhimmel** [comment]
Closing this issue as part of a routine cleaning. People can still use this issue as a reference and keep discuss below. However, I don't think there are any unresolved action items related to this issue.
